### PR TITLE
py/stream: Use MicroPython errno instead of system errno.h.

### PR DESCRIPTION
**Pull request by aykevl**
This makes compatibility with Clang a bit better and seems a bit more correct to me.

### CODE REVIEW

File: py/stream.c [hunk 1]
```diff
@@ -33,7 +33,7 @@
 #include "py/runtime.h"
 
 #if MICROPY_STREAMS_NON_BLOCK
-#include <errno.h>
+#include "py/mperrno.h"
 #if defined(__MINGW32__) && !defined(__MINGW64_VERSION_MAJOR)
 #define EWOULDBLOCK 140
 #endif
```
Comment (dpgeorge) [inline py/stream.c]: I think this entire block can now be removed because EWOULDBLOCK is never needed, and mperrno.h is included by stream.h.

**Comment by dpgeorge** [comment]
Thanks for the patch, I did look at doing this myself some time ago but didn't because it may introduce some subtle issues.

The point is that MP_EWOULDBLOCK is defined as EAGAIN for systems that are not using uPy's internal errnos.  And so the behaviour of the `mp_is_nonblocking_error()` macro will change with this patch.  The way to fix that is to change the definition of MP_EWOULDBLOCK in py/mperrno.h to define it as the system's EWOULDBLOCK.  And arguably that is the right thing to do so I made a PR #3741 for that change.

But regardless of that issue, the patch here seems correct to me because it is more consistent with use of errno's.  Eg it may be that a stream returns MP_EAGAIN but the `mp_is_nonblocking_error()` macro doesn't catch this value because it checks for EAGAIN instead (which may have a different value than MP_EAGAIN when MICROPY_USE_INTERNAL_ERRNO is enabled).

File: py/stream.c (hunk 1 above)
Comment (aykevl) [reply py/stream.c]: You're right, it's not needed. I've removed the entire block.

**Comment by dpgeorge** [comment]
Merged in d43c7377564b3e1be36f627a465df2c5779a1ae8

**Pull request closed without merging**